src/Fhir.Anonymizer.Core/Extensions/FhirPathSymbolExtensions.cs

```diff
@@ -7,10 +7,21 @@ namespace Fhir.Anonymizer.Core.Extensions
 {
     public static class FhirPathSymbolExtensions
     {
+        private static object _lock = new object();
         public static SymbolTable AddExtensionSymbols(this SymbolTable t)
         {
-            t.Add("nodesByType", (IEnumerable<ITypedElement> f, string typeName) => NodesByType(f, typeName), doNullProp: true);
-            t.Add("nodesByName", (IEnumerable<ITypedElement> f, string name) => NodesByName(f, name), doNullProp: true);
+            lock (_lock)
```

Comment:
What is this change used for?

Reply:
We add the lock for thread-safety reasons as extension methods are added to FHIRCompiler.defaultsymboltable which is a static List varible. Adding to the list from different threads will cause possible enumeration failures.